cloud/src/meta-service/txn_lazy_committer.cpp

```diff
@@ -0,0 +1,208 @@
+                        sub_partition_tmp_rowset_metas(tmp_rowset_metas.begin() + i,
+                                                       tmp_rowset_metas.begin() + end);
+                convert_tmp_rowsets(instance_id_, txn_id_, txn_kv_, code_, msg_, db_id,
+                                    sub_partition_tmp_rowset_metas, tablet_ids);
+                if (code_ != MetaServiceCode::OK) break;
+            }
+            if (code_ != MetaServiceCode::OK) break;
+
+            DCHECK(tmp_rowset_metas.size() > 0);
+            std::unique_ptr<Transaction> txn;
+            TxnErrorCode err = txn_kv_->create_txn(&txn);
+            if (err != TxnErrorCode::TXN_OK) {
+                code_ = cast_as<ErrCategory::CREATE>(err);
+                ss << "failed to create txn, txn_id=" << txn_id_ << " err=" << err;
+                msg_ = ss.str();
+                LOG(WARNING) << msg_;
+                break;
+            }
+
+            int64_t table_id = -1;
+            for (auto& [tmp_rowset_key, tmp_rowset_pb] : tmp_rowset_metas) {
+                if (table_id <= 0) {
+                    table_id = tablet_ids[tmp_rowset_pb.tablet_id()].table_id();
+                }
+                txn->remove(tmp_rowset_key);
+            }
+
+            DCHECK(table_id > 0);
+            DCHECK(partition_id > 0);
+
+            std::string ver_val;
+            std::string ver_key =
+                    partition_version_key({instance_id_, db_id, table_id, partition_id});
+            err = txn->get(ver_key, &ver_val);
+            if (TxnErrorCode::TXN_OK != err) {
+                code_ = err == TxnErrorCode::TXN_KEY_NOT_FOUND ? MetaServiceCode::TXN_ID_NOT_FOUND
+                                                               : cast_as<ErrCategory::READ>(err);
+                ss << "failed to get partiton version, txn_id=" << txn_id_
+                   << " key=" << hex(ver_key) << " err=" << err;
+                msg_ = ss.str();
+                LOG(WARNING) << msg_;
+                break;
+            }
+            VersionPB version_pb;
+            if (!version_pb.ParseFromString(ver_val)) {
+                code_ = MetaServiceCode::PROTOBUF_PARSE_ERR;
+                ss << "failed to parse version pb txn_id=" << txn_id_ << " key=" << hex(ver_key);
+                msg_ = ss.str();
+                break;
+            }
+
+            if (version_pb.has_txn_id() && version_pb.txn_id() == txn_id_) {
+                version_pb.clear_txn_id();
+                ver_val.clear();
+                if (!version_pb.SerializeToString(&ver_val)) {
+                    code_ = MetaServiceCode::PROTOBUF_SERIALIZE_ERR;
+                    ss << "failed to serialize version_pb when saving, txn_id=" << txn_id_;
+                    msg_ = ss.str();
+                    return;
+                }
+                txn->put(ver_key, ver_val);
+            }
+
+            err = txn->commit();
```

Comment:
No need to commit if `version_pb.has_txn_id() && version_pb.txn_id() == txn_id_` is false.

Reply:
> No need to commit if `version_pb.has_txn_id() && version_pb.txn_id() == txn_id_` is false.

done